modules/hcp-vm-client/templates/setup.sh

```diff
@@ -17,8 +17,7 @@ setup_deps () {
     version="${consul_version}"
     consul_package="consul-enterprise="$${version:1}"*"
     apt install -qy apt-transport-https gnupg2 curl lsb-release nomad $${consul_package} getenvoy-envoy unzip jq apache2-utils nginx
-    curl -fsSL https://get.docker.com -o get-docker.sh
-    sh ./get-docker.sh
+    sudo apt install docker docker.io -y
```

Comment:
Sweet and thanks for opening this. Do you mind linking to more info about when/why this convenience script was deprecated? I still see it on the Docker documentation: https://docs.docker.com/engine/install/ubuntu/#install-using-the-convenience-script

Reply:
Thanks, I've just try to find scripts on github from official documentation to propose the change because I had some automation that use the same script and all of them started to fail a couple of days ago.

this is a log from a EC2 in AWS:
Ubuntu 16.04.7 LTS (GNU/Linux 4.4.0-1112-aws x86_64)

<img width="1292" alt="Captura de Pantalla 2022-05-20 a las 18 12 45" src="https://user-images.githubusercontent.com/11272068/169569383-1bfebb7f-815a-4e3f-9dfb-0585e382f5ae.png">


I hope that this help,

Reply:
My fault because of the version of the ubuntu,